plugins/woocommerce/src/Internal/Abilities/REST/RestAbilityFactory.php

```
@@ -0,0 +1,259 @@
+	}
+
+
+	/**
+	 * Check permissions - delegate to REST controller.
+	 *
+	 * @param object $controller REST controller instance.
+	 * @param string $operation Operation type.
+	 * @param array  $input Input parameters.
+	 * @return bool Whether permission is granted.
+	 */
+	private static function check_permission( $controller, string $operation, array $input ): bool {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$controller'. (undefined)

(UnusedFormalParameter)